### ISSUE TITLE
Agentless cannot be used on Kail

### ISSUE THREAD

**Issue by libccat** [wazuh/wazuh#13381]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.2 | wazuh-agentless | Manager | Packages| centos7 |

`Agentless` connection to Ubuntu is normal, but there is a problem connecting to Kail.

use `/var/ossec/bin/wazuh-agentlessd -fdd` , Kail and Ubuntu show the same information.
see `/var/ossec/logs/ossec.log`, Ubuntu normal scan with `fim_db_search()`:
![image](https://user-images.githubusercontent.com/33114231/167091124-eac6facf-475a-4dfd-a999-604867237491.png)
Kail just quit:
![image](https://user-images.githubusercontent.com/33114231/167090819-b8754895-efb5-4bf0-abdb-2ab016c44b88.png)

**Comment by vikman90** [comment]
Hi @libccat,

I'm trying to reproduce this issue, but the host seems to work properly. I'm using the Vagrant box [kalilinux/rolling](https://app.vagrantup.com/kalilinux/boxes/rolling).

This is my configuration:
```xml
<agentless>
  <type>ssh_integrity_check_linux</type>
  <frequency>36000</frequency>
  <host>root@192.168.182.136</host>
  <state>periodic</state>
  <arguments>/etc</arguments>
</agentless>
```

And this is my log:

<details><summary>Agentlessd logs</summary>

```
2022/05/06 11:23:34 wazuh-agentlessd[5688] debug_op.c:70 at _log(): DEBUG: Logging module auto-initialized
2022/05/06 11:23:34 wazuh-agentlessd[5688] main.c:106 at main(): DEBUG: Wazuh home directory: /var/ossec
2022/05/06 11:23:34 wazuh-agentlessd[5688] main.c:152 at main(): DEBUG: Chrooted to directory: /var/ossec, using user: wazuh
2022/05/06 11:23:34 wazuh-agentlessd[5688] main.c:165 at main(): INFO: Started (pid: 5688).
2022/05/06 11:23:36 wazuh-agentlessd[5688] mq_op.c:46 at StartMQ(): DEBUG: Connected succesfully to 'queue/sockets/queue' after 0 attempts
2022/05/06 11:23:36 wazuh-agentlessd[5688] mq_op.c:47 at StartMQ(): DEBUG: (unix_domain) Maximum send buffer set to: '212992'.
2022/05/06 11:23:36 wazuh-agentlessd[5688] pthreads_op.c:45 at CreateThreadJoinable(): DEBUG: Thread stack size set to: 8192 KiB
2022/05/06 11:23:36 wazuh-agentlessd[5688] lessdcom.c:77 at lessdcom_main(): DEBUG: Local requests thread ready
2022/05/06 11:23:36 wazuh-agentlessd[5688] agentlessd.c:364 at run_periodic_cmd(): INFO: Test passed for 'ssh_integrity_check_linux'.

2022/05/06 11:24:37 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer: spawn ssh root@192.168.182.136
2022/05/06 11:24:37 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer:
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer: Linux kali 5.15.0-kali3-amd64 #1 SMP Debian 5.15.15-2kali1 (2022-01-31) x86_64
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer:
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer: The programs included with the Kali GNU/Linux system are free software;
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer: the exact distribution terms for each program are described in the
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer: individual files in /usr/share/doc/*/copyright.
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer:
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer: Kali GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer: permitted by applicable law.
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer: Last login: Fri May  6 05:18:20 2022 from 192.168.182.1
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer:
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:390 at run_periodic_cmd(): INFO: ssh_integrity_check_linux: root@192.168.182.136: Started.
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer: unset HISTFILE; echo "INFO: Starting."; for i in `find /etc 2>/dev/null`;do tail $i >/dev/null 2>&1 && md5=`md5sum $i | cut -d " " -f 1` && sha1=`sha1sum $i | cut -d " " -f 1` && echo FWD: `stat --printf "%s:%a:%u:%g" $i`:$md5:$sha1 $i; done; exit
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer: exit
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer:
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer: └─# unset HISTFILE; echo "INFO: Starting."; for i in `find /etc 2>/dev/null`;do
2022/05/06 11:24:38 wazuh-agentlessd[5688] agentlessd.c:390 at run_periodic_cmd(): INFO: ssh_integrity_check_linux: root@192.168.182.136: Starting.

2022/05/06 11:24:55 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer: Connection to 192.168.182.136 closed.
2022/05/06 11:24:55 wazuh-agentlessd[5688] agentlessd.c:410 at run_periodic_cmd(): DEBUG: Buffer:
2022/05/06 11:24:55 wazuh-agentlessd[5688] agentlessd.c:390 at run_periodic_cmd(): INFO: ssh_integrity_check_linux: root@192.168.182.136: Finished.
```

</details>

I see two main differences from your log:
1. I've got no Python message paragraph. However, I'm not sure this is the cause of the issue.
2. Your log shows a syntax error as commands `unset` and `echo` are not comma-separated:
```
unset HISTFILE echo "INFO: Starting."; (...)
```
Instead of:
```
unset HISTFILE; echo "INFO: Starting."; (...)
```

This is a known issue (#9200), fixed by #9292 (4.3.0, released yesterday). I confirm that the Agentless script picks the files and delivers them to Analysisd:

<details><summary>Analysisd logs</summary>

```
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:378 at fim_db_search(): DEBUG: Agent '000' File '/etc/ipp-usb/ipp-usb.conf'
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:379 at fim_db_search(): DEBUG: Agent '000' Old checksum '1632:644:0:0:bece66e39470a768d351068c1a43c080:64aba445186e83ac71ebf4173ed7dfc91f36c428::::::'
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:380 at fim_db_search(): DEBUG: Agent '000' New checksum '1632:644:0:0:bece66e39470a768d351068c1a43c080:64aba445186e83ac71ebf4173ed7dfc91f36c428'
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:477 at fim_db_search(): DEBUG: Agent '000' File /etc/ipp-usb/ipp-usb.conf saved/updated in FIM DDBB
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:1124 at fim_get_scantime(): DEBUG: Agent '000' FIM end_scan '1651828622'
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:507 at fim_db_search(): DEBUG: Agent '000' End end_scan is '1651828622' (lf->time: '1651829078')
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:378 at fim_db_search(): DEBUG: Agent '000' File '/etc/screenrc'
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:379 at fim_db_search(): DEBUG: Agent '000' Old checksum '3663:644:0:0:12c245238eb8b653625bba27dc81df6a:1acc7d1e6222696e95346d32e337d9619e398c70::::::'
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:380 at fim_db_search(): DEBUG: Agent '000' New checksum '3663:644:0:0:12c245238eb8b653625bba27dc81df6a:1acc7d1e6222696e95346d32e337d9619e398c70'
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:477 at fim_db_search(): DEBUG: Agent '000' File /etc/screenrc saved/updated in FIM DDBB
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:507 at fim_db_search(): DEBUG: Agent '000' End end_scan is '1651828622' (lf->time: '1651829078')
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:378 at fim_db_search(): DEBUG: Agent '000' File '/etc/bash_completion'
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:379 at fim_db_search(): DEBUG: Agent '000' Old checksum '45:644:0:0:a81b3f1cb197219b815942f4fc7fa94e:7c54123ec53be7fcc7a69f75d7182184d4d602fd::::::'
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:380 at fim_db_search(): DEBUG: Agent '000' New checksum '45:644:0:0:a81b3f1cb197219b815942f4fc7fa94e:7c54123ec53be7fcc7a69f75d7182184d4d602fd'
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:477 at fim_db_search(): DEBUG: Agent '000' File /etc/bash_completion saved/updated in FIM DDBB
2022/05/06 11:24:38 wazuh-analysisd[5529] syscheck.c:507 at fim_db_search(): DEBUG: Agent '000' End end_scan is '1651828622' (lf->time: '1651829078')
(...)
```

</details>

In order to troubleshoot this problem, try to run the command that you see in the log directly on Kali Linux:
```shell
unset HISTFILE; echo "INFO: Starting."; for i in `find /etc 2>/dev/null`;do tail $i >/dev/null 2>&1 && md5=`md5sum $i | cut -d " " -f 1` && sha1=`sha1sum $i | cut -d " " -f 1` && echo FWD: `stat --printf "%s:%a:%u:%g" $i`:$md5:$sha1 $i; done; exit
```

<details><summary>Result sample</summary>

```
INFO: Starting.
FWD: 1632:644:0:0:bece66e39470a768d351068c1a43c080:64aba445186e83ac71ebf4173ed7dfc91f36c428 /etc/ipp-usb/ipp-usb.conf
FWD: 3663:644:0:0:12c245238eb8b653625bba27dc81df6a:1acc7d1e6222696e95346d32e337d9619e398c70 /etc/screenrc
FWD: 45:644:0:0:a81b3f1cb197219b815942f4fc7fa94e:7c54123ec53be7fcc7a69f75d7182184d4d602fd /etc/bash_completion
FWD: 260:644:0:0:1a2b9d0a869e2aa885ae3621c557fb95:452357e5e22571aacf9f4dd1ae2bca9767851257 /etc/dpkg/shlibs.default
FWD: 253:644:0:0:84b1e69080569cc5c613a50887af5200:99885863fe6a0cd39ac0f26b9ba5610805a66092 /etc/dpkg/shlibs.override
FWD: 446:644:0:0:f4413ffb515f8f753624ae3bb365b81b:17c8e76036626a55e95f91dca7f60ac4dc027bb5 /etc/dpkg/dpkg.cfg
FWD: 91:644:0:0:3ec487a080323ca00648685e2f1030d5:d597f4010902067d0daee5eb565307bdffc459ea /etc/dpkg/origins/kali
FWD: 4:777:0:0:3ec487a080323ca00648685e2f1030d5:d597f4010902067d0daee5eb565307bdffc459ea /etc/dpkg/origins/default
FWD: 83:644:0:0:c47b6815f67ad1aeccb0d4529bd0b990:06fe380fb5d1f3e55d4925c181602b3aa2f80317 /etc/dpkg/origins/debian
FWD: 206:644:0:0:6e4c2dde6c6970f124de9493e2c286fa:0d395df29e51a3cc2104c59abd9820a292cddf36 /etc/idmapd.conf
FWD: 17097:644:0:0:ca17dde03d50b99eef7630148bc8d8d9:e8d5bc44b481fbcf1cefbcd8899cb3211bf41150 /etc/wireshark/init.lua
FWD: 1468:755:0:0:b75b20ba3fd750b33e47dd81e2c427a8:c6fc7cd8091517c3e3256b9942f5a34460c7013c /etc/openvpn/update-resolv-conf
FWD: 890:644:0:0:c358838e1789c1d4e6da7f525fc922cf:576ffc2f578a8b78b2295584c059338d976c485a /etc/gtk-3.0/im-multipress.conf
FWD: 63:644:0:0:637ddabcc79d883e234653873dac9c84:4902573bfaa4fe40cd3169a23368212ca6849a59 /etc/resolv.conf
FWD: 3287:644:0:0:17f23c6a0494e3ee9e23de54996ac6dd:f0e1ca4eab4cd428f0e51140fda753f3d64ab8bf /etc/nikto.conf
FWD: 2839:755:0:0:4e71ee082fe21cbefd8ccd27a69c5f7e:9d019123e0c17c74090c4c1860575513de45e1a2 /etc/responder/Responder.conf
FWD: 711:644:0:0:208d7a7756d6daa71111bb0296f4ce6a:56dbe6039c7ad3c6479054cc74cafe6e03c4a78e /etc/hosts.deny
FWD: 509:644:0:0:fd8518f7c253ca493803426dad732fac:b5d1673e1bbcfafebbfa67429ce5bc52a660a70a /etc/reader.conf.d/libccidtwin
FWD: 1107:644:0:0:b3d9e29a36f945b2065b2f88d18dadaa:8885907baec3899ef96764a08bbe053455224b76 /etc/profile.d/gawk.csh
FWD: 726:644:0:0:4257431f99f10e5fbe86b39b6b9e5002:dd8e5f7156c585759b8c3d4d30b4ddf80f0911fb /etc/profile.d/bash_completion.sh
FWD: 1908:644:0:0:a1f60728d39e70bdfb1039b296dd52ca:ddc83bfdb4ff88b6258ad17e680678fce30d68ee /etc/profile.d/vte-2.91.sh
FWD: 390:644:0:0:48acdcc50e7bc8e44b6796db23f31015:4d3f16ddb4966cc4bc783183ac37fc24af63bc83 /etc/profile.d/kali.sh
FWD: 967:644:0:0:1f922824a6d29c466178d0834d17321c:2c7c9d0a2ffb8d98a5bdace2dfc8f1603aa0ecf5 /etc/profile.d/vte.csh
FWD: 39:644:0:0:ebf0e217bbc4f419d454a33386ed78d0:abb42949723beb10ee2618359e4e2c5264ee18d8 /etc/profile.d/dotnet-cli-tools-bin-path.sh
FWD: 757:644:0:0:bfc054c0862d0fad98ca641b951c7061:e69dfc5c1b719471f65c11b88325b80ae7e373e7 /etc/profile.d/gawk.sh
FWD: 0:644:0:0:d41d8cd98f00b204e9800998ecf8427e:da39a3ee5e6b4b0d3255bfef95601890afd80709 /etc/subgid-
FWD: 426:600:0:0:a4b53d2f0dae261dee832602847a788b:fcd71010f653d18c60259193d09eb3130cb67980 /etc/vpnc/default.conf
FWD: 30:644:0:0:8b362e210615e66b3bf7f69f6c819056:db982378758922a2f7cbf810904cf66d587e71a3 /etc/issue
FWD: 1585:440:0:0:0c96b3c8db0a3fc9b1405fd54899fc11:d7f97f43444463519f98f0b91c8fbcd97b78faa3 /etc/sudoers
FWD: 281:644:0:0:cf3d1b642d68b1af384aa0d8adf31deb:0ab0ffe3e3b646ac7a57c2cf784cf4d5f61d8d14 /etc/strongswan.conf
FWD: 1816:644:0:0:cde059c510632569fb1869eb86cc026d:2902538bf5531857042f2b3aff36b076fd384759 /etc/ethertypes
FWD: 116:644:0:0:761b638c6f1e83219ca8fc3d63953aba:67261566d8d82090003ad3bf1cd6f6e09486e080 /etc/opensc/opensc.conf
FWD: 825:644:0:0:187f9a1d7072dc853e14e9bf64219d7c:2adfe9908b4da5ca31c1ca7c49bac390e98d4a09 /etc/powershell-empire/server/config.yaml
FWD: 2632:644:0:0:7834d359c63e9d731fd7050f82804cb1:6e6fe02d6f7d5d001a4f73b26d21229dcb8b86e1 /etc/powershell-empire/client/config.yaml
FWD: 604:644:0:0:773fb95e98a27947de4a95abb3d3f2a2:40e84d9b448fe5f603cecb7e0d03f28cf7c244c8 /etc/deluser.conf
FWD: 10055:644:0:0:88c01374e4d28459e5b1ded894daaead:3ce3865b0eeb92d5ba462dc36113fca66a02ba0d /etc/ettercap/etter.conf
(...)
exit
```

</details>

On the other hand, give a try with 4.3.0, which fixed the bug above.

Hope that helps.

Best regards.

**Comment by libccat** [comment]
I discussed this problem with the staff on Slack before. 
At that time, I used `agentless` to connect Kail. I saw that it stopped working, and I said there might be a problem with `;` .
But he said that without `;` , it would not affect the use of `agentless`, and there was no problem with his computer.
Then I used it to Ubuntu and saw that it does not affect the use of `agentless`.

So I raised this question, now I know the reason according to your answer. Thank you very much.
And when I run this script with `;` directly in Kail, it actually works normally as I expected.

However, you can see that there is no `;` on Ubuntu, `agentless` also works normally. Why? Is it just an accident that it can't run on Kail?
